src/main/java/hudson/remoting/Channel.java

```diff
@@ -829,8 +830,16 @@ public void terminate(IOException e) {
             } // JENKINS-14909: leave synch block
         } finally {
             if (e instanceof OrderlyShutdown) e = null;
-            for (Listener l : listeners.toArray(new Listener[0]))
-                l.onClosed(this, e);
+            for (Listener l : listeners.toArray(new Listener[0])) {
+                try {
+                    l.onClosed(this, e);
+                } catch (Throwable t) {
```

Comment:
Debatable. e.g. if it was an OutOfMemory error it may be that closing this channel will resolve the issue anyway. Now we could do the

```
} catch (RuntimeException e) { 
  // log
} catch (Error e) {
  throw e;
} catch (Throwable t) {
  // log
}
```
dance if you feel really strongly about it... though all that will do is propagate the Error to Channel#L487 where it gets caught ant logged (incorrectly) as a failure to create the close command